vault/logical_system.go

```diff
@@ -3182,6 +3186,19 @@ func (b *SystemBackend) handleWrappingLookup(ctx context.Context, req *logical.R
 		Path:        "cubbyhole/wrapinfo",
 		ClientToken: token,
 	}
+
+	te, err := b.Core.tokenStore.lookupTokenNonLocked(ctx, token, false)
+	if err != nil {
+		return nil, err
+	}
+	if te == nil {
+		return logical.ErrorResponse("invalid token"), logical.ErrInvalidRequest
+	}
+
+	cubbyReq.TokenEntryVersion = te.Version
+
+	cubbyReq.SetTokenEntry(te)
```

Comment:
Isn't the line just above this not necessary, given that you're passing both the req and the te into the function? Seems like in all the cases you're doing this, simply moving that first line into the function would be cleaner.

Reply:
Fixed.